src/Spice86.Core/Emulator/InterruptHandlers/Dos/DosInt21Handler.cs

```diff
@@ -89,6 +103,54 @@ public class DosInt21Handler : InterruptHandler {
         _dispatchTable.Add(0x62, new Callback(0x62, GetPspAddress));
     }
 
+    public void TerminateProgram() {
+        if (_loggerService.IsEnabled(LogEventLevel.Verbose)) {
+            _loggerService.Verbose("Program requested exit via INT2F AH=0x00 (Terminate Program). Terminating now");
+        }
+        _cpu.IsRunning = false;
+    }
+
+    public void BufferedInput() {
+        
+    }
+
+    /// <summary>
+    /// Waits for a character from standard input, echoes it to standard output, and returns the character in the AL register. <br/>
+    /// Echoes on standard output.
+    /// </summary>
+    public void CharacterInputWithEcho() {
+        // Standard input is not implemented, we return no character ready.
+        _state.AL = 0;
+    }
+    
+    /// <summary>
+    /// Waits for a character from standard input, echoes it to standard output, and returns the character in the AL register.
+    /// </summary>
+    public void CharacterInputWithoutEcho() {
+        // Standard input is not implemented, we return no character ready.
+        _state.AL = 0;
+    }
+
+    /// <summary>
+    /// Waits for a character from the standard auxiliary device and returns the character int he AL register.
+    /// </summary>
+    public void AuxiliaryInput() {
+        // NOP
+        _state.AL = 0;
+    }
+    
+    /// <summary>
+    /// Sends a character in _state.DL to the first AUX device.
+    /// </summary>
+    public void AuxiliaryOutput() {
+        // NOP
```

Comment:
This should not be a NOP. 
It should either write the character to AUX (we have an AUX device: `Spice86.Core.Emulator.OperatingSystem.Dos.AddDefaultDevices()` and an STDAUX filehandle: `Spice86.Core.Emulator.OperatingSystem.Dos.OpenDefaultFileHandles`),
or log a warning that we're ignoring the call,
or do nothing and not handle it.